File: leetcode.com/3432_count_partitions_with_even_sum_difference/solution.py
"""
Sum balance

TC: O(N)
SC: (1)
"""
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        total = 0
        left, right = 0, sum(nums)

        for i in range(len(nums) - 1):
            left += nums[i]
            right -= nums[i]

            if (left - right) % 2 == 0:
                total += 1

        return total

File: leetcode.com/3432_count_partitions_with_even_sum_difference/solution.c
int countPartitions(int *nums, int numsSize)
{
    int left = 0;
    int right = 0;

    for (int i = 0 ; i < numsSize ; i++)
        right += nums[i];

    int total = 0;
    for (int i = 0 ; i < numsSize - 1 ; i++) {
        left += nums[i];
        right -= nums[i];

        if ((right - left) % 2 == 0) {
            total++;
        }
    }

    return total;
}
